   # Winter 2021 Data Science Intern Challenge 

   ## Question1

In [ ]:
On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 

1.Think about what could be going wrong with our calculation. Think about a better way to evaluate this data.  \
2.What metric would you report for this dataset?\
3.What is its value?


In [36]:
import pandas as pd

In [37]:
#import the google doc file
sheet_id = '16i38oonuX1y1g7C_UAmiK9GkY7cS-64DfiDMNiR41LM'
df=pd.read_csv(f'https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=csv')
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [38]:
df.dtypes

order_id           int64
shop_id            int64
user_id            int64
order_amount       int64
total_items        int64
payment_method    object
created_at        object
dtype: object

In [39]:
df['order_amount'].sum()/df['order_id'].count()

3145.128

The reason for this value is there is an outlier in this dataset, there is a order that total amount is 70400 and 
the total item is 2000 pairs. This causes the AOV extremely high. To evaluate this data set, use the order_amount/total_items is a better
way to report this dataset

In [40]:
df['price_per_pair']=df['order_amount']/df['total_items']

In [41]:
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  price_per_pair  
0  2017-03-13 12:36:56           112.0  
1  2017-03-03 17:38:52            90.0  
2   2017-03-14 4:23:56           144.0  
3  2017-03-26 12:43:37           156.0  
4   2017-03-01 4:35:11           156.0

In [42]:
df['price_per_pair'].mean()

387.7428

After the calculation, the value 387.74 is more convincible for sneakers

## Question 2

For this question you’ll need to use SQL. Follow this link to access the data set required for the challenge. Please use queries to answer the following questions. Paste your queries along with your final numerical answers below.

1.How many orders were shipped by Speedy Express in total?\
2.What is the last name of the employee with the most orders?\
3.What product was ordered the most by customers in Germany?



1. 54 in total \
   SELECT COUNT(O.OrderID)\
   FROM Orders O,Shippers S\
   WHERE S.ShipperID = O.ShipperID AND S.ShipperName = 'Speedy Express'

2.  Last Name:Peacock \
    SELECT E.LastName, COUNT(O.OrderID) \
    FROM Orders O, Employees E\
    WHERE E.EmployeeID = O.EmployeeID\
    GROUP BY E.LastName\
    ORDER BY COUNT(O.OrderID) DESC

3. Product:Gorgonzola Telino\
    SELECT P.ProductName, COUNT(O.OrderID) FROM Products P, Orders O, OrderDetails D, Customers C\
    WHERE P.ProductID = D.ProductID AND O.OrderID = D.OrderID AND O.CustomerID = C.CustomerID \
    AND C.Country = 'Germany'\
    GROUP BY P.ProductName\
    ORDER BY COUNT(O.OrderID) DESC


